korangar/src/world/library/mod.rs

```diff
@@ -164,7 +197,98 @@ impl Library {
         Ok(compacted)
     }
 
-    fn load_map_sky_data_table(state: &Lua) -> mlua::Result<HashMap<String, MapSkyData>> {
+    fn load_skill_list_table(state: Lua) -> mlua::Result<HashMap<SkillId, SkillListEntry>> {
+        let globals = state.globals();
+        let mut result = HashMap::new();
+
+        if let Ok(table) = globals.get::<mlua::Table>("SKILL_INFO_LIST") {
+            for (skill_id, table) in table.pairs::<u16, mlua::Table>().flatten() {
+                let file_name = table.get(1)?;
+                let name = table.get("SkillName")?;
+                let maximum_level = table.get("MaxLv")?;
+
+                let generic_required_skills = match table.get::<mlua::Table>("_NeedSkillList") {
+                    Ok(sequence) => {
+                        let mut required_skills = HashMap::new();
+
+                        for required_skill in sequence.sequence_values::<mlua::Table>() {
+                            let required_skill = required_skill?;
+                            let skill_id = required_skill.get::<u16>(1)?;
+                            // TODO: At least one skill does not have a level set. I am just
+                            // assuming that that means level 1 is required but should be
+                            // confirmed.
+                            let skill_level = required_skill.get::<u16>(2).unwrap_or(1);
+                            required_skills.insert(SkillId(skill_id), SkillLevel(skill_level));
+                        }
+
+                        required_skills
+                    }
+                    Err(..) => HashMap::new(),
+                };
+
+                let job_required_skills = match table.get::<mlua::Table>("NeedSkillList") {
+                    Ok(table) => {
+                        let mut job_specific = HashMap::new();
+
+                        for (job_id, sequence) in table.pairs::<u16, mlua::Table>().flatten() {
+                            let mut required_skills = HashMap::new();
+
+                            for required_skill in sequence.sequence_values::<mlua::Table>() {
+                                let required_skill = required_skill?;
+                                let skill_id = required_skill.get::<u16>(1)?;
+                                // TODO: At least one skill does not have a level set. I am just
```

Comment:
Feel free to disregard if this isn't intended to remain in the code: I'd always add which skill (example/ID/name) so that others don't need to search/run the code/wonder if the comment is outdated should they encounter it later.

(Even better, if maybe excessive: Programmatically encode assumptions and let the machine detect invalid ones)